### PR TITLE
Check GG proxy for client assignment. 

### PR DESCRIPTION
**Pull request by bravesirrobin81**

### CODE REVIEW

File: app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala [hunk 1]
```diff
@@ -23,19 +42,32 @@ class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost) {
...
-    val agentCode = (agentDetails \ "AgentCode").head.text
+    if (agentDetails.nonEmpty) {
+      val agentCode = (agentDetails \ "AgentCode").head.text
 
-    val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
-      AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
+        AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      }
+      Some(AgentDetails(agentCode, credentials))
```
Comment (szmg) [inline app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala]: Based on the top level element (`AllocatedAgents`) there can be more agents, which case is not handled here.

```
	<AllocatedAgents>
		<AgentDetails>
			<AgentCode>123456ABCDEF</AgentCode>
			<AssignedCredentials>
				<Credential>
					<CredentialIdentifier>001</CredentialIdentifier>
				</Credential>
			</AssignedCredentials>
		</AgentDetails>
		<AgentDetails>
			<AgentCode>123456789ABC</AgentCode>
			<AssignedCredentials>
				<Credential>
					<CredentialIdentifier>007</CredentialIdentifier>
				</Credential>
			</AssignedCredentials>
		</AgentDetails>
	</AllocatedAgents>
```

would return `Some(AgentDetails("123456ABCDEF", Seq("001", "007"))`, while agent 007 does not belong to that agency.

File: app/uk/gov/hmrc/agentaccesscontrol/service/GovernmentGatewayAuthorisationService.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.Future
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+
+class GovernmentGatewayAuthorisationService(val ggProxyConnector: GovernmentGatewayProxyConnector) {
+
+  def isAuthorisedInGovernmentGateway(saUtr: SaUtr, ggCredentialId: String)(implicit hc: HeaderCarrier): Future[Boolean] = {
+    ggProxyConnector.getAssignedSaAgents(saUtr) map {
+      case Some(agentDetails) if agentDetails.assignedCredentials.head.identifier == ggCredentialId => true
```
Comment (szmg) [inline app/uk/gov/hmrc/agentaccesscontrol/service/GovernmentGatewayAuthorisationService.scala]: There can be multiple assigned credentials, check them as well.

File: app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala (hunk 1 above)
Comment (bravesirrobin81) [inline app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala]: It was a fairly conscious decision to not handle this case. We don't know what the shape of the data is, or if it's possible from a business POV. A question definitely to ask the GG team. I suggest we defer any changes until we know what the data look like.

Comment (szmg) [inline app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala]: We only don't know the business POV, all the others suggest it's a list:

XSD is `<xs:element maxOccurs="unbounded" name="AgentDetails" type="AgentDetailsType" />`.

Alos, GG doc quite unambiguously claims that it returns a list of agent codes, check part 3.3.1.27.

**Pull request merged**